### PR TITLE
feat: add urdu language intl

### PR DESCRIPTION
**Pull request by MuhammadAnas190**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [x] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Added Urdu language (ur_PK) locale     |
| 🇨🇳 Chinese |     添加了乌尔都语 (ur_PK) 语言环境      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/31346#issuecomment-877828142

File: components/locale-provider/__tests__/index.test.js [hunk 1]
```diff
@@ -140,6 +141,7 @@ const locales = [
   zhCN,
   zhHK,
   zhTW,
+  urPK,
```
Comment (yoyo837) [inline components/locale-provider/__tests__/index.test.js:144]: @afc163 为啥这个不会引起snapshot变化？

Comment (afc163) [reply components/locale-provider/__tests__/index.test.js:144]: 会，不过是新增，所以跑 npm test 不会报错。

**Comment by afc163** [comment]
Please run `npm test -- components/locale-provider/__tests__/index.test.js` and commit the snapshot change.

**Comment by MuhammadAnas190** [comment]
@afc163 updated the PR content and snapshot. Please check

**Review by yoyo837 — APPROVED** [review]

**Pull request merged**